sunpy/net/attr.py

```diff
@@ -243,20 +227,28 @@ def update_values(cls, adict):
                             if len(pair) == 1:
                                 # Special case handling for * aka all values allowed.
                                 if pair[0] == "*":
-                                    strtonum
                                     pair = ["all", "All values of this type are supported."]
                                 else:
                                     raise ValueError(
                                         f'Invalid value given for * registration: {attr_values}.')
                             else:
                                 raise ValueError(f'Invalid length (!=2) for values: {attr_values}.')
-                        # Sanitize part one: Check if the first letter is a number and
-                        # replace it with a string version
-                        if NUMBER_REGEX.match(pair[0][0]):
-                            # This turns that digit into its name
-                            name = strtonum(pair[0][0])
-                            if pair[0][1:]:
-                                name = name + "_" + pair[0][1:]
+                        p = inflect.engine()
+                        # Sanitize part one: Check if the name is has a number in it
+                        if NUMBER_REGEX.match(pair[0]):
+                            # Now check if the entire name is a number
+                            if len(pair[0]) == NUMBER_REGEX.match(pair[0]).span()[1]:
+                                # This turns that digit into its name
```

Comment:
```suggestion
                                # This turns that number into its name
```
Are we converting a digit (i.e. the firs one?) into a name or the whole number?

Reply:
The entire number here.
Second if is just the digit.